sign/ed25519/ed25519_test.go

```diff
@@ -39,6 +39,24 @@ func TestMalleability(t *testing.T) {
 	}
 }
 
+func TestPublic(t *testing.T) {
+	var zero zeroReader
+	pub, priv, err := ed25519.GenerateKey(zero)
+	if !priv.Equal(priv) {
+		t.Fatal()
+	}
+	if !pub.Equal(pub) {
+		t.Fatal()
+	}
+	if err != nil {
```

Comment:
Why is the error check not the first assertion?

Reply:
That's a bug. Thanks for pointing that out.